tests/deploy_test.go

```diff
@@ -174,107 +144,59 @@ func Test_Deploy_WithAnnotations(t *testing.T) {
 		EnvVars:     envVars,
 	}
 
-	deployStatus, deployErr := deploy(t, functionRequest)
-	if deployErr != nil {
-		t.Errorf(deployErr.Error())
-		t.Fail()
-		return
-	}
-
+	deployStatus := deploy(t, functionRequest)
 	if deployStatus != http.StatusOK && deployStatus != http.StatusAccepted {
-		t.Logf("got %d, wanted %d or %d", deployStatus, http.StatusOK, http.StatusAccepted)
-		t.Fail()
+		t.Fatalf("got %d, wanted %d or %d", deployStatus, http.StatusOK, http.StatusAccepted)
 	}
 
-	invoke(t, functionRequest.Service, emptyQueryString, http.StatusOK)
+	_ = invoke(t, functionRequest.Service, emptyQueryString, http.StatusOK)
 	function := get(t, functionRequest.Service)
 
 	if err := strMapEqual("annotations", *function.Annotations, wantedAnnotations); err != nil {
-		t.Log(err)
-		t.Fail()
+		t.Fatal(err)
 	}
 }
 
-func deploy(t *testing.T, createRequest requests.CreateFunctionRequest) (int, error) {
-	gwURL, urlErr := url.Parse(os.Getenv("gateway_url"))
-
-	if urlErr != nil {
-		t.Fatal(urlErr)
-	}
-
-	gwURL.Path = "/system/functions"
-
-	body, res, err := httpReq(gwURL.String(), http.MethodPost, makeReader(createRequest))
-
-	if err != nil {
-		return http.StatusBadGateway, err
-	}
+func deploy(t *testing.T, createRequest requests.CreateFunctionRequest) int {
```

Comment:
We could just remove the dependency on `testing.T`, I think functions that return an error vs Panic/Fatal mid-flow are more predictable / re-usable. What's the intent in the change?